src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php

```diff
@@ -94,6 +94,8 @@ protected function addFactory(ContainerBuilder $container, MetadataInterface $me
         $factoryClass = $metadata->getClass('factory');
         $modelClass = $metadata->getClass('model');
 
+        $container->setAlias($factoryClass, $metadata->getServiceId('factory'));
```

Comment:
What would happen if there are two resources which use the same factory class for them? Could you include that scenario in the test suite and see what happens?

Reply:
@pamil The second factory alias replace the first one. I've just added a test to check that.

Reply:
Just to bring this discussion on Slack here, we figured out the solution to be named autowiring like `Factory $bookFactory` and `Factory $authorFactory`, not `Factory $whateverFactory`.